prow/bugzilla/client.go

```diff
@@ -127,29 +129,37 @@ func (c *client) GetBug(id int) (*Bug, error) {
 	return parsedResponse.Bugs[0], nil
 }
 
-func getClones(c Client, bug *Bug) ([]*Bug, error) {
+func getClones(c Client, bug *Bug, cache map[int]*Bug) ([]*Bug, error) {
 	var errs []error
 	clones := []*Bug{}
 	for _, dependentID := range bug.Blocks {
-		dependent, err := c.GetBug(dependentID)
-		if err != nil {
-			errs = append(errs, fmt.Errorf("Failed to get dependent bug #%d: %v", dependentID, err))
-			continue
+
+		dependent, ok := cache[dependentID]
```

Comment:
`continue` on cache hit rather than branching on cache miss

Reply:
Same case as above, If i were to continue on cache hit I wouldn't execute the next if loop